runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3146,6 +3146,22 @@ remoteCompile(
    if (compiler->isOptServer())
       compiler->setOption(TR_Server);
    auto classInfoTuple = JITServerHelpers::packRemoteROMClassInfo(clazz, compiler->fej9vm()->vmThread(), compiler->trMemory());
+
+   // check if the romClass is in the map, if it is not insert in the map
+   // otherwise send an empty string 
+   if (compInfo->getRomJ9ClassMap().find(romClass) == compInfo->getRomJ9ClassMap().end())
+      {
+      // rom class was not found send the romclass
+      // send_compile_request_with_romclass
+      compInfo->getRomJ9ClassMap().insert({romClass, classInfoTuple});
+      }
+   else
+      {
+      // send an empty string instead of the rom class   
+      // send_compile_request_without_romclass 
+      std::get<0>(classInfoTuple) = " ";
```

Comment:
I wonder if we can optimize this path because we seem to serialize the ROMClass information and then throw it away